webapp/lib/view.dart

```diff
@@ -500,161 +475,186 @@ class BatchRepliesConfigurationView extends PackageConfiguratorView {
         ..append(addsTagsContainer)
     );
   }
+}
+
+class TagListView extends BaseView {
+  List<TagView> tagElements;
+  DivElement _tagsContainer;
+  SpanElement _tagsActionContainer;
+  Function onTagChangedCallback;
 
-  SpanElement _addTagAction(List<String> tags) {
-    return new SpanElement()
+  TagListView(this.tagElements, Map<String, model.TagStyle> tagList, this.onTagChangedCallback, [bool tagsEditable = false]) {
+    _tagsContainer = new DivElement()
+      ..classes.add('tags');
+    _tagsActionContainer = new SpanElement()
       ..classes.add('tags__actions')
       ..append(
         new ButtonElement()
           ..classes.add('button-add-tag')
           ..text = '+'
           ..onClick.listen((event) {
-            var tagActions = (event.target as Element).parent;
-            var tagList = tagActions.parent;
-            _addTagDropdown(tagActions, tagList, model.tags);
+            if (tagsEditable) {
+              _tagsActionContainer.insertAdjacentElement('beforebegin', new TagView('', model.TagStyle.Normal, onTagChangedCallback, tagsEditable).renderElement);
+            } else {
+              _addTagDropdown(tagList, onTagChangedCallback);
+            }
             event.stopPropagation();
```

Comment:
A couple of suggestions here:
* prefer to call `stopPropagation` as the first action in the event listener
* as a general suggestion: treat special cases as special. In this case it would mean to prefer an early exit on one of the if branches instead of an if/else construction for a binary choice. It's easy to forget then when the special case ends, and you might end up doing an action after the if/else block that is only relevant for one of the branches, so resulting in pretty obscure bugs if that happens (not the case here, just guarding against the future). An alternative would be to use a switch, but it's too much hassle when it's a binary choice.

```suggestion
            event.stopPropagation();
            if (tagsEditable) {
              _tagsActionContainer.insertAdjacentElement('beforebegin', new TagView('', model.TagStyle.Normal, onTagChangedCallback, tagsEditable).renderElement);
              return;
            }
            _addTagDropdown(tagList, onTagChangedCallback);
```

Reply:
Noted